cont.c

```diff
@@ -1909,6 +1909,20 @@ rb_cont_call(int argc, VALUE *argv, VALUE contval)
     UNREACHABLE_RETURN(Qnil);
 }
 
+static void
+fiber_handle_weak_references(void *ptr)
+{
+    rb_fiber_t *fiber = ptr;
+
+    if (!fiber) return;
+
+    if (!rb_gc_handle_weak_references_alive_p(fiber->cont.saved_ec.gen_fields_cache.obj) ||
+            !rb_gc_handle_weak_references_alive_p(fiber->cont.saved_ec.gen_fields_cache.fields_obj)) {
+        fiber->cont.saved_ec.gen_fields_cache.obj = Qundef;
+        fiber->cont.saved_ec.gen_fields_cache.fields_obj = Qundef;
+    }
+}
+
```

Comment:
Why moving this to the continuation part before the fiber document, not just before `rb_fiber_data_type`?